jctools-core/src/main/java/org/jctools/queues/atomic/MpmcAtomicArrayQueue.java

```diff
@@ -72,17 +97,56 @@ protected final boolean casConsumerIndex(long expect, long newValue) {
     }
 }
 
+/**
+ * NOTE: This class was automatically generated by org.jctools.queues.atomic.JavaParsingAtomicArrayQueueGenerator
+ * which can found in the jctools-build module. The original source file is MpmcArrayQueue.java.
+ */
 abstract class MpmcAtomicArrayQueueL3Pad<E> extends MpmcAtomicArrayQueueConsumerIndexField<E> {
+
     long p01, p02, p03, p04, p05, p06, p07;
+
     long p10, p11, p12, p13, p14, p15, p16, p17;
+
     public MpmcAtomicArrayQueueL3Pad(int capacity) {
         super(capacity);
     }
 }
 
+/**
+ * NOTE: This class was automatically generated by org.jctools.queues.atomic.JavaParsingAtomicArrayQueueGenerator
+ * which can found in the jctools-build module. The original source file is MpmcArrayQueue.java.
```

Comment:
The above is prepended to the original JavaDoc so that atomics variant also has this documentation easily accessible but a warning first that the class is generated.